src/API/Status/PreviewCard.vala

```diff
@@ -121,6 +129,47 @@ public class Tuba.API.PreviewCard : Entity, Widgetizable {
 		}
 	}
 
+	#if CLAPPER_0_10
+		private bool enhancer_check (string scheme, string? t_host = null) {
+			if (!Clapper.WITH_ENHANCERS_LOADER) return false;
+			bool supported = false;
+
+			string? host = t_host;
+			if (host != null && host.has_prefix ("www.")) host = host.substring (4);
+
+			var proxies = Clapper.get_global_enhancer_proxies ();
+			for (var i = 0; i < proxies.get_n_proxies (); i++) {
+				var proxy = proxies.get_proxy (i);
+
+				string? proxy_schemes = proxy.get_extra_data ("X-Schemes");
+				if (proxy_schemes == null || proxy_schemes == "") continue;
+
+				string[] schemes = proxy_schemes.split (";");
+				if (!(scheme in schemes)) continue;
+
+				// We were only looking for scheme matching
+				// e.g. peertube://
+				if (host == null) {
+					supported = true;
+					break;
+				}
+
+				string? proxy_hosts = proxy.get_extra_data ("X-Hosts");
+				if (proxy_hosts == null || proxy_hosts == "") continue;
+
+				string[] hosts = proxy_hosts.split (";");
+				// both the host and the scheme
+				// match at this point
+				if (host in hosts) {
+					supported = true;
+					break;
+				}
```

Comment:
Same as above, this can be probably simplified too:

```vala
if (proxy.extra_data_lists_value ("X-Hosts", host)) {
  supported = true;
  break;
}
```

I will update documentation later, with mention that its safe to call this even with "key" does not exist.